version: '3.1'

services:

  jupyter:
    image: jupyter_environment
    restart: always
    user: root
    environment:
      - JUPYTER_ENABLE_LAB=yes
      - NB_UID=1001
      - NB_GID=1001
#      - NB_UID=$$(id -u)
#      - NB_GID=$$(id -g)
      - GRANT_SUDO=yes
    volumes:
      - ./src:/home/jovyan/work/src
      - ./notebooks:/home/jovyan/work/notebooks
      - ./data/external:/home/jovyan/work/data/external
      - ./data/interim:/home/jovyan/work/data/interim
      - ./data/processed:/home/jovyan/work/data/processed
      - ./data/raw:/home/jovyan/work/data/raw
      - ./reports:/home/jovyan/work/reports
    ports:
      - "8888:8888"
    networks:
      - my-network

networks:
  my-network:
